source/Octopus.Tentacle.Client/TentacleClient.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Halibut;
+using Halibut.ServiceModel;
+using Octopus.Tentacle.Client.ClientServices;
+using Octopus.Tentacle.Client.Decorators;
+using Octopus.Tentacle.Client.Retries;
+using Octopus.Tentacle.Client.Scripts;
+using Octopus.Tentacle.Contracts;
+using Octopus.Tentacle.Contracts.Capabilities;
+using Octopus.Tentacle.Contracts.ScriptServiceV2;
+using Polly.Timeout;
+using ILog = Octopus.Diagnostics.ILog;
+
+namespace Octopus.Tentacle.Client
+{
+    public class TentacleClient : ITentacleClient
+    {
+        readonly IScriptObserverBackoffStrategy scriptObserverBackOffStrategy;
+        readonly RpcCallRetryHandler rpcCallRetryHandler;
+        readonly IClientScriptService scriptServiceV1;
+        readonly IClientScriptServiceV2 scriptServiceV2;
+        readonly IClientFileTransferService fileTransferServiceV1;
+        readonly IClientCapabilitiesServiceV2 capabilitiesServiceV2;
+
+        public TentacleClient(ServiceEndPoint serviceEndPoint,
+            IHalibutRuntime halibutRuntime,
+            IScriptObserverBackoffStrategy scriptObserverBackOffStrategy,
+            ITentacleServiceDecorator? tentacleServicesDecorator,
```

Comment:
If the main reason for having this decorator is observability, then could we expose something like `OnRemovtCall` event and decouple it from the services that Tentacle Client uses behind the scenes? 

E.g. https://github.com/OctopusDeploy/OctopusClients/blob/master/source/Octopus.Server.Client/IOctopusAsyncClient.cs#L43-L48

Reply:
As discussed we will revisit this after this PR merges to hide the Service Decorator from the client and keep it internal to the Tentacle solution